x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/SpatialPluginTests.java

```diff
@@ -35,8 +35,8 @@ public void testGeoCentroidLicenseCheck() {
             List<Consumer<ValuesSourceRegistry.Builder>> registrar = plugin.getAggregationExtentions();
             registrar.forEach(c -> c.accept(registryBuilder));
             ValuesSourceRegistry registry = registryBuilder.build();
-            GeoCentroidAggregatorSupplier centroidSupplier = (GeoCentroidAggregatorSupplier) registry.getAggregator(
-                new ValuesSourceConfig(GeoShapeValuesSourceType.instance(), null, false, null, null, null),
+            MetricAggregatorSupplier centroidSupplier = (MetricAggregatorSupplier) registry.getAggregator(
+                new ValuesSourceConfig(GeoShapeValuesSourceType.instance(), null, true, null, null, null, null, null, null),
```

Comment:
This is a super long number of arguments! I'm glad we don't call the ctor very much though.

Reply:
Yeah, outside of testing, the only thing that should be calling that constructor is the various resolve factory methods, which are what the rest of the framework should call.  Not that they don't have a lot of arguments too.  I suppose we could have a default constructor for tests, but I don't like adding production code that's only supposed to be called from tests.

Reply:
How often do we call it in tests? It doesn't look like we call it directly very often. I think mostly we use the production resolution framework.

Reply:
Correct, there's only a handful of cases we call the constructor from tests.  I meant I didn't want to add a new test-only constructor just to clean up this case.

Reply:
I'm happy with that, yeah.